bundles/net.tourbook.tests/test/utils/Comparison.java

```diff
@@ -0,0 +1,129 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import net.tourbook.common.util.StringUtils;
+import net.tourbook.data.TourData;
+
+import org.skyscreamer.jsonassert.ArrayValueMatcher;
+import org.skyscreamer.jsonassert.Customization;
+import org.skyscreamer.jsonassert.JSONCompare;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.skyscreamer.jsonassert.JSONCompareResult;
+import org.skyscreamer.jsonassert.comparator.CustomComparator;
+
+public class Comparison {
+
+   private static final String JSON = ".json"; //$NON-NLS-1$
+
+   /**
+    * Compares a test transaction against a control transaction.
+    *
+    * @param testTourData
+    *           The generated test TourData object.
+    * @param controlFileName
+    *           The control's file name.
+    */
+   public static void CompareTourDataAgainstControl(final TourData testTourData,
+                                                    final String controlFileName) {
+
+      // When using Java 11, convert the line below to the Java 11 method
+      //String controlDocument = Files.readString(controlDocumentFilePath, StandardCharsets.US_ASCII);
+
+      final String controlDocumentFilePath = Paths.get(controlFileName + JSON).toAbsolutePath().toString();
+      final String controlDocument = readFile(controlDocumentFilePath, StandardCharsets.US_ASCII);
+
+      final String testJson = testTourData.toJson();
+
+      final ArrayValueMatcher<Object> arrValMatch = new ArrayValueMatcher<>(new CustomComparator(
+            JSONCompareMode.LENIENT,
+            new Customization("tourMarkers[*].deviceLapTime", (o1, o2) -> true), //$NON-NLS-1$
+            new Customization("tourMarkers[*].tourData", (o1, o2) -> true))); //$NON-NLS-1$
+
+      final Customization arrayValueMatchCustomization = new Customization("tourMarkers", arrValMatch); //$NON-NLS-1$
+      final CustomComparator customArrayValueComparator = new CustomComparator(
+            JSONCompareMode.LENIENT,
+            arrayValueMatchCustomization,
+            new Customization("tourId", (o1, o2) -> true), //$NON-NLS-1$
+            new Customization("startTimeOfDay", (o1, o2) -> true)); //$NON-NLS-1$
+
+      final JSONCompareResult result =
+            JSONCompare.compareJSON(controlDocument, testJson, customArrayValueComparator);
+
+//      if (result.failed()) {
+//         WriteErroneousFiles(controlFileName, testJson);
+//      }
+
+      assert result.passed();
+   }
+
+   private static String readFile(final String path, final Charset encoding) {
+
+      if (StringUtils.isNullOrEmpty(path)) {
+         return UI.EMPTY_STRING;
+      }
+
+      byte[] encoded = null;
+      try {
+         encoded = Files.readAllBytes(Paths.get(path));
+         return new String(encoded, encoding);
+      } catch (final IOException e) {
+         e.printStackTrace();
+      }
+
+      return UI.EMPTY_STRING;
+   }
+
+   public static TourData RetrieveImportedTour(final HashMap<Long, TourData> newlyImportedTours) {
```

Comment:
same naming above

Reply:
Fixed in my [PR](https://github.com/wolfgang-ch/mytourbook/pull/255)